src/chainstate/stacks/boot/bns.clar

```diff
@@ -0,0 +1,973 @@
+      (owned-name (map-get? owner-name ((owner contract-caller)))))
+    ;; The sender principal must match the namespace's import principal
+    (asserts!
+      (is-eq (get namespace-import namespace-props) contract-caller)
+      (err ERR_NAMESPACE_OPERATION_UNAUTHORIZED))
+    ;; The name's namespace must not be launched
+    (asserts!
+      (is-none (get launched-at namespace-props))
+      (err ERR_NAMESPACE_ALREADY_LAUNCHED))
+    ;; Less than 1 year must have passed since the namespace was "revealed"
+    (asserts!
+      (< block-height (+ (get revealed-at namespace-props) NAMESPACE_LAUNCHABILITY_TTL))
+      (err ERR_NAMESPACE_PREORDER_LAUNCHABILITY_EXPIRED))
+    ;; Update eventual name-props
+    (if (is-none owned-name)
+      true
+      (let ((name (unwrap! (get name owned-name) (err ERR_PANIC))))
+        (let ((name-props (unwrap! (map-get? name-properties ((name name) (namespace namespace))) (err ERR_PANIC))))
+          ;; Unset imported-at, Set registered-at
+          (map-set name-properties
+            ((name name) (namespace namespace))
+            ((registered-at (some block-height))
+            (imported-at none)
+            (revoked-at none)
+            (zonefile-hash (get zonefile-hash name-props)))))))
+    ;; The namespace will be set to "launched"
+    (map-set namespaces
+      ((namespace namespace))
+      ((launched-at (some block-height))
+       (namespace-import (get namespace-import namespace-props))
+       (revealed-at (get revealed-at namespace-props))
+       (namespace-version (get namespace-version namespace-props))
+       (renewal-rule (get renewal-rule namespace-props))
+       (price-function (get price-function namespace-props))))
+    (ok true)))
+
+;;;; NAMES
+
+;; NAME_PREORDER
+;; This is the first transaction to be sent. It tells all BNS nodes the salted hash of the BNS name,
+;; and it pays the registration fee to the namespace owner's designated address
+(define-public (name-preorder (hashed-salted-fqn (buff 20))
+                              (stx-to-burn uint))
+  (let 
+    ((former-preorder 
+      (map-get? name-preorders ((hashed-salted-fqn hashed-salted-fqn) (buyer contract-caller)))))
+    ;; Ensure eventual former pre-order expired 
+    (asserts! 
+      (if (is-none former-preorder)
+        true
+        (>= block-height (+ NAME_PREORDER_CLAIMABILITY_TTL
+                            (unwrap! (get created-at former-preorder) (err ERR_PANIC)))))
+      (err ERR_NAME_PREORDER_ALREADY_EXISTS))
+          (asserts! (> stx-to-burn u0) (err ERR_NAMESPACE_STX_BURNT_INSUFFICIENT))    
+    ;; Ensure that the hashed fqn is 20 bytes long
+    (asserts! (is-eq (len hashed-salted-fqn) u20) (err ERR_NAME_HASH_MALFORMED))
+    ;; Ensure that user will be burning a positive amount of tokens
+    (asserts! (> stx-to-burn u0) (err ERR_NAME_STX_BURNT_INSUFFICIENT))
+    ;; Burn the tokens
+    (unwrap! (stx-transfer? stx-to-burn contract-caller BURN_ADDRESS) (err ERR_INSUFFICIENT_FUNDS))
```

Comment:
If this is a version-3 namespace, then the STX should be sent to the namespace creator's address instead of burned.

Reply:
I think @kantai would like to get rid of this namespace version. Any take on that approach?

Reply:
Hey, sorry for the late reply.  I think it's fine to get rid of it; I just want to make sure we're all in agreement.